dirty_cat/datetime_encoder.py

```diff
@@ -0,0 +1,173 @@
+        The "other" feature will be a numerical value expressed in the "extract_until" unit.
+    add_day_of_the_week: bool, default=False
+        Add day of the week feature (if day is extracted). This is a numerical feature from 0 to 6.
+    add_holidays : bool, default=False
+        Whether to add a numerical variable encoding if the day of the date is a holiday (1 for holiday,
+        0 for non-holiday).
+        Uses pandas calendar, which for now only contains US holidays.
+
+    Attributes
+    ----------
+    n_features_out: int
+        Number of features of the transformed data.
+    features_per_column: Dict[int, List[str]]
+        A dictionary mapping the index of the original columns
+        to the list of features extracted for each column.
+    """
+    def __init__(self,
+                 extract_until="hour",
+                 add_day_of_the_week=False,
+                 add_holidays=False):
+        to_extract = ["year", "month", "day", "hour", "minute", "second", "millisecond", "microsecond", "nanosecond"]
+        self.extract_until = extract_until
+        self._validate_keywords()
+        self.to_extract_full = to_extract[:to_extract.index(extract_until) + 1]
+        self.to_extract_full.append("other")
+        self.add_day_of_the_week = add_day_of_the_week
+        self.add_holidays = add_holidays
+        # Some functions need aliases
+        self.word_to_alias = {"year": "Y", "month": "M", "day": "D", "hour": "H", "minute": "min", "second": "S",
+                              "millisecond": "ms", "microsecond": "us", "nanosecond": "N"}
+
+    def _validate_keywords(self):
+        if self.extract_until not in ["year", "month", "day", "hour", "minute", "second", "millisecond", "microsecond", "nanosecond"]:
+            msg = (
+                """extract_until should be one of ["year", "month", "day", "hour", "minute", "second", "millisecond", 
+                "microsecond", "nanosecond"], got {0}.""".format(
+                    self.extract_until
+                )
+            )
+            raise ValueError(msg)
+
+    def _extract_from_date(self, date_series, feature):
+        if feature == "year":
+            return pd.DatetimeIndex(date_series).year.to_numpy()
+        elif feature == "month":
+            return pd.DatetimeIndex(date_series).month.to_numpy()
+        elif feature == "day":
+            return pd.DatetimeIndex(date_series).day.to_numpy()
+        elif feature == "hour":
+            return pd.DatetimeIndex(date_series).hour.to_numpy()
+        elif feature == "minute":
+            return pd.DatetimeIndex(date_series).minute.to_numpy()
+        elif feature == "second":
+            return pd.DatetimeIndex(date_series).second.to_numpy()
+        elif feature == "millisecond":
+            return pd.DatetimeIndex(date_series).microsecond.to_numpy()
+        elif feature == "microsecond":
+            return pd.DatetimeIndex(date_series).microsecond.to_numpy()
```

Comment:
I don't get why it returns the same exact same thing when specifying "millisecond" and "microsecond", is there a reason ?

Reply:
Oooops